config.md

```diff
@@ -399,6 +403,20 @@ The [state](runtime.md#state) of the container MUST be passed to hooks over stdi
 The pre-start hooks MUST be called after the [`start`](runtime.md#start) operation is called but [before the user-specified program command is executed](runtime.md#lifecycle).
 On Linux, for example, they are called after the container namespaces are created, so they provide an opportunity to customize the container (e.g. the network namespace could be specified in this hook).
 
+Note: Prestart hooks were deprecated in favor of mount and start hooks, which have a clearer definition.
```

Comment:
I would explicitly mention that `runc` incorrectly implemented them, to ensure users don't depend on them.

Reply:
Are we going to use -host and -container or -inside and -outside suffixes to the hook names? It's still not clear to me if proposed hooks will be executed in the host or container namespace. "mount" looks quite vague to me in this regard. It requires some guesswork to understand its purpose. Would it be better to have prestart-host and prestart-container instead?

Reply:
@cyphar @Renaudwastaken  Can you clarify the statement about "`runc` incorrectly implemented them" ?  From what I see in https://github.com/opencontainers/runtime-spec/blob/master/config.md#posix-platform-hooks it says "Hooks MUST be executed in the runtime namespace.", and  prestart is executed accordingly to this specification.

Reply:
> The pre-start hooks MUST be called after the start operation is called but before the user-specified program command is executed. 

This points to having the hook after the `pivot_root` is called :)

https://github.com/opencontainers/runtime-spec/blob/master/config.md#prestart

Reply:
@bart0sh, is the naming something that can be resolved by better detailling the hooks in the spec?

Today they are worded as follows:
*mount hooks*
> The mount hooks MUST be called as part of the create operation at a time that would allow the hook to execute operations both inside and outside the container.
> For example, on Linux this would happen before the `pivot_root` operation is executed but after the mount namespace was created and setup.

*start hooks*
> The start hooks MUST be called [before the user-specified process is executed](runtime.md#lifecycle) as part of the [`start`](runtime.md#start) operation.
> This hook can be used to execute some operations in the container, for example running the ldconfig binary on linux before the container process is spawned.

Do you think we could reword the definition in a different way that would reduce the confusion?

Reply:
> > The pre-start hooks MUST be called after the start operation is called but before the user-specified program command is executed.
> 
> This points to having the hook after the `pivot_root` is called :)
> https://github.com/opencontainers/runtime-spec/blob/master/config.md#prestart

Well, not really. It specifies the timing. But where hook itself is called is specified by line that I quoted from spec above: "Hooks MUST be executed in the runtime namespace", and runtime namespace defined as  https://github.com/opencontainers/runtime-spec/blob/master/glossary.md#runtime-namespace

"After `pivot_root`" will be already not a runtime namespace.